examples/gno.land/r/demo/users/users.gno

```diff
@@ -7,44 +7,16 @@ import (
 	"strings"
 
 	"gno.land/p/demo/avl"
+	"gno.land/p/demo/users"
 )
 
-//----------------------------------------
-// Types
-
-type User struct {
-	address std.Address
-	name    string
-	profile string
-	number  int
-	invites int
-	inviter std.Address
-}
-
-func (u *User) Render() string {
-	str := "## user " + u.name + "\n" +
-		"\n" +
-		" * address = " + string(u.address) + "\n" +
-		" * " + strconv.Itoa(u.invites) + " invites\n"
-	if u.inviter != "" {
-		str = str + " * invited by " + string(u.inviter) + "\n"
-	}
-	str = str + "\n" +
-		u.profile + "\n"
-	return str
-}
-
-func (u User) Name() string         { return u.name }
-func (u User) Profile() string      { return u.profile }
-func (u User) Address() std.Address { return u.address }
-
 //----------------------------------------
 // State
 
 var (
 	admin      std.Address = "g1us8428u2a5satrlxzagqqa5m6vmuze025anjlj"
-	name2User  avl.Tree                    // Name -> *User
-	addr2User  avl.Tree                    // std.Address -> *User
+	name2User  avl.Tree                    // Name -> *users.User
```

Comment:
A user storage structure seems like a great candidate for something that should be included in a users package.  It makes it easy for applications to maintain and access their own user list.  Something like:
```
type p/users.Storage struct {
    nameToUser *avl.Tree
    addresstoUser *avl.Tree
}

func (s *Storage) GetUserByName(name string) *User
func (s *Storage) GetExtendeUserByName(name string) *ExtendedUser
func (s *Storage) getUserByName(name string) interface{}{ return s.nameToUser.Get(name) }

etc
...
```